CONTRIBUTING.md

```diff
@@ -1,44 +1,184 @@
 
-### Step 1: Fork
+### Asking for General Help
 
-Fork the project [on GitHub](https://github.com/nodejs/node) and clone your fork
-locally.
+Because the level of activity in the `nodejs/node` repository is so high,
+questions or requests for general help using Node.js should be directed at
+the [Node.js help repository][].
 
-```text
-$ git clone git@github.com:username/node.git
-$ cd node
-$ git remote add upstream https://github.com/nodejs/node.git
+### Discussing non-technical topics
+
+Discussion of non-technical topics (such as intellectual property and trademark)
+should be directed to the [Technical Steering Committee (TSC) repository][].
+
+### Submitting a Bug Report
+
+When opening a new issue in the `nodejs/node` issue tracker, users will be
+presented with a basic template that should be filled in.
+
+```markdown
+<!--
+Thank you for reporting an issue.
+
+This issue tracker is for bugs and issues found within Node.js core.
+If you require more general support please file an issue on our help
+repo. https://github.com/nodejs/help
+
+
+Please fill in as much of the template below as you're able.
+
+Version: output of `node -v`
+Platform: output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)
+Subsystem: if known, please specify affected core module name
+
+If possible, please provide code that demonstrates the problem, keeping it as
+simple and free of external dependencies as you are able.
+-->
+
+* **Version**:
+* **Platform**:
+* **Subsystem**:
+
+<!-- Enter your issue details below this comment. -->
 ```
 
-#### Which branch?
+If you believe that you have uncovered a bug in Node.js, please fill out this
+form, following the template to the best of your ability. Do not worry if you
+cannot answer every detail, just fill in what you can.
+
+The two most important pieces of information we need in order to properly
+evaluate the report is a description of the behavior you are seeing and a simple
+test case we can use to recreate the problem on our own. If we cannot recreate
+the issue, it becomes impossible for us to fix.
+
+In order to rule out the possibility of bugs introduced by userland code, test
+cases should be limited, as much as possible, to using *only* Node.js APIs.
+If the bug occurs only when you're using a specific userland module, there is
+a very good chance that either (a) the module has a bug or (b) something in
+Node.js changed that broke the module.
+
+### Triaging a Bug Report
+
+Once an issue has been opened, it is not uncommon for there to be discussion
+around it. Some contributors may have differing opinions about the issue,
+including whether the behavior being seen is a bug or a feature. This discussion
+is part of the process and should be kept focused, helpful and professional.
 
-For developing new features and bug fixes, the `master` branch should be pulled
-and built upon.
+Short, clipped responses--that provide neither additional context nor supporting
```

Comment:
Are these double dashes intended (+ in the next line)?

Reply:
Yeah, stand in for an emdash. I should make those actual dashes

Reply:
Should they be rendered as emdash by GitHub or they are OK by convention? Sorry, I do not want to bother anyone needlessly for this in the future)